mcstatus/pinger.py

```diff
@@ -219,36 +225,30 @@ def __init__(self, raw):
 
         if "description" not in raw:
             raise ValueError("Invalid status object (no 'description' value)")
-        if isinstance(raw["description"], (dict, list)):
-            if isinstance(raw["description"], dict):
-                entries = raw["description"].get("extra", ())
-                end = raw["description"]["text"]
-            else:
-                entries = raw["description"]
-                end = ""
-
-            description = ""
+        self.description = self._parse_description(raw["description"])
 
-            for entry in entries:
-                if entry.get("bold"):
-                    description += "§l"
-
-                if entry.get("italic"):
-                    description += "§o"
+        self.favicon = raw.get("favicon")
 
-                if entry.get("underlined"):
-                    description += "§n"
```

Comment:
Were there some test cases that you used to test this? It should be straightforward to test a few dictionary inputs in the unit test. Would you mind adding those?

Reply:
Done in c3ebf12, (which actually did uncover an issue in the implementation, which I've then fixed in 
8cdcf6b).

However I think there may be an issue with this implementation. This does follow the same logic as the original implementation did so if this really was an issue, it was there originally too. The thing I'm talking about is for example something like this:
```py
        out = PingResponse._parse_description(
            {
                "extra": [
                    {
                        "bold": True,
                        "italic": True,
                        "color": "gray",
                        "text": "foo"
                    },
                    {
                        "color": "gold",
                        "text": "bar"
                    }
                ],
                "text": "."
            }
        )
        # We don't care in which order the style prefixes are, allow any
        assert out in {
            "§l§o§7foo§6bar.",
            "§l§7§ofoo§6bar.",
            "§o§l§7foo§6bar.",
            "§o§7§lfoo§6bar.",
            "§7§o§lfoo§6bar.",
            "§7§l§ofoo§6bar.",
        }
```
What I'm unsure about here is the fact that for example the ending `"text"` probably shouldn't have any style, yet we never reset the colors, same goes for the text in the second extra group, it probably shouldn't be affected by the style of the first one, but we never reset the color after going to another group. We just add on the style/colors we found in that next group, not worrying about any style resets, should we?

Reply:
I'll open an issue to remind me to look at adding test cases for fancy statuses, but this refactoring that you have done should have the same behavior as the existing code. The while loop might be unnecessary complexity since it only goes one level in the "color" part of the style map, but I'm alright with that. It could be useful for nesting more types of styles in the future :man_shrugging:

Reply:
Do you have an example minecraft server that provides their MOTD/description back in this dictionary format? I tried this out on a vanilla 1.18.1 server with an MOTD of `\u00A7l\u00A77bold and gray\u00A7rreset` and the `.status().raw["description"]` had `§l§7bold and gray§rreset`. Which Minecraft servers return a dictionary here instead of the string?

Reply:
> IDo you have an example minecraft server that provides their MOTD/description back in this dictionary format?

It took me a while to find one too, but `2b2t.org` did return a dict like this.